google/cloud/bigquery/dataset.py

```diff
@@ -763,6 +764,25 @@ def default_encryption_configuration(self, value):
             api_repr = value.to_api_repr()
         self._properties["defaultEncryptionConfiguration"] = api_repr
 
+    @property
+    def storage_billing_model(self):
+        """Union[str, None]: StorageBillingModel of the dataset as set by the user
+        (defaults to :data:`None`).
+
+        Raises:
+            ValueError: for invalid value types.
+        """
+        return self._properties.get("storageBillingModel")
+
+    @storage_billing_model.setter
+    def storage_billing_model(self, value):
+        if not isinstance(value, str) and value is not None:
+            raise ValueError("Pass a string, or None")
```

Comment:
We're missing some unit tests for some of these branches:

```
google/cloud/bigquery/dataset.py                               352      2     81      3    99%   780, 781->783, 784
```

Reply:
any ETA on this ?